localenv/cloud-nine-wallet/docker-compose.yml

```diff
@@ -78,6 +79,13 @@ services:
     depends_on:
       - shared-database
       - shared-redis
+    healthcheck:
+      test: ["CMD", "wget", "--header=apollo-require-preflight: true", "http://localhost:3001/graphql?query=%7B__typename%7D", "-O", "/dev/null"]
+      start_period: 60s
+      start_interval: 5s
+      interval: 5m
+      retries: 10
+      timeout: 3s
```

Comment:
Some things to note here:

First, the healthcheck does _**not**_ stop after it succeeds. I noticed this after seeing lots of logs for this gql operation after startup and successful seeding ([a SO post observing the same thing](https://devops.stackexchange.com/questions/13463/nginx-healthcheck-does-not-stop)).

This is different than I expected and why I specified the start period/interval. This checks more upfront then backs off after 60s (and maybe after the first success?) I think the start period probably needs to be greather than `start_interval` * `retries` (ensures that the start is unhealthy within the `start_period`).

Second, we use `wget` because there is no curl, and the endpoint is [what apollo sever suggests for a healthcheck](https://www.apollographql.com/docs/apollo-server/monitoring/health-checks/). [I initially added](https://github.com/interledger/rafiki/pull/2911/commits/1316e74dcb23d809e20d82925cd26708129a0600#diff-5b481c1a1464d3e282df7102a0cb5882701fc2b2af81037e203812e13acf0176L376-L378) a new `/healthz` path to the koa server which we mount the apollo server to but figured the apollo server itself was a better, more direct check.

Reply:
Is it a log like this? to check that its working?

```
{"level":30,"time":1726148987820,"pid":42,"hostname":"cloud-nine-wallet-backend","requestId":"53de7860-30f7-46bb-8548-aca479dfb9de","operation":null}

```

Reply:
So it keeps making the requests even if its the start_period?

Reply:
yeah thats the log.

> So it keeps making the requests even if it succeeded the first time during the start_period?

and yes it keeps making the requests

30s would be fine, just figured we didn't really care about it after the initial one gating the MASEs (hence made it much longer).

Reply:
In that case, we can just reduce the # of retries? maybe 30s interval, 5 times?

Reply:
what does that accomplish? after it succeeds it will still send the healtcheck for each interval. lowering the retries just makes the threshold lower for sending out a signal that it failed to start

Reply:
Is it possible that we don't have any successful starts during the `start_period`, and then we have to wait 5 mins for the next check?

Reply:
Normally no because we're doing 10 retries and checking every 5s (50s + time to make the http request for each) but looking at it again, yes maybe. 

The start interval is 5s and the retries are 10 so within the 60s start period it should fail or succeed if the duration of each healthcheck is not >2s. Which should normally be the case (if its not up its an ECONREFUSED and it fails fast - doesn't hang). Currently the timeout is 3s though, so if it hung for 3s each time I think that could happen.

Reply:
I bumped the start period to 90s. so even if all 10 healthchecks take 8 seconds (interval + timeout) then it will fail within the start period.

Reply:
Looks like the number of retries are not considered during the start up time ([doc](https://docs.docker.com/reference/dockerfile/#healthcheck)):

> start period provides initialization time for containers that need time to bootstrap. Probe failure during that period will not be counted towards the maximum number of retries.

This means there will be 10 retries after the start up time of 90s, so we just need to determine how long it should take after the startup time to keep trying the check